lib/Sema/MiscDiagnostics.cpp

```diff
@@ -1401,10 +1401,16 @@ static void diagSyntacticUseRestrictions(const Expr *E, const DeclContext *DC,
       Expr *subExpr = nullptr;
       if (calleeName == "??" &&
           (subExpr = isImplicitPromotionToOptional(lhs))) {
+
+        auto removeStart =
+          Lexer::getLocForEndOfToken(Ctx.SourceMgr, lhs->getEndLoc());
+        auto removeEnd =
+          Lexer::getLocForEndOfToken(Ctx.SourceMgr, rhs->getEndLoc());
+
         Ctx.Diags.diagnose(DRE->getLoc(), diag::use_of_qq_on_non_optional_value,
                            subExpr->getType())
           .highlight(lhs->getSourceRange())
-          .fixItRemove(SourceRange(DRE->getLoc(), rhs->getEndLoc()));
+          .fixItRemoveChars(removeStart, removeEnd);
```

Comment:
Did this affect existing tests? If not can you add test cases that exercise the situation we are addressing?
If you are unfamiliar with the testing, see exemples of testing here https://github.com/apple/swift/blob/1122050658e10151795d02bd960ea58ee7f9ec00/test/Constraints/casts.swift#L251

Reply:
@LucianoPAlmeida Thank you for your reply.

I have also tried to write test cases, but I couldn't find a suitable way to test the "fix action".  In this case, the warning message pointing to start of "??" is as expected, but the automatic fix operation (i.e., `fixItRemoveChars`) is expected to start deleting from the character immediately after the first non-empty character on the left side of "??" and continue until the end of "rhs".

Reply:
> but I couldn't find a suitable way to test the "fix action".

Fix-it actions for removal are tested with `{{1-2=}}` where 1 is start column and 2 end column so it would look something like this `/// expected-error {{message}} {{1-2=}}` so we would  verify if the fix-it removed the whitespace range as well. 
See here for more reference: https://github.com/apple/swift/blob/main/docs/Diagnostics.md#diagnostic-verifier

Reply:
This should probably affect existing tests already. Have you try to run them locally? 
If not see how to do it here https://github.com/apple/swift/blob/main/docs/HowToGuides/GettingStarted.md#running-tests

Reply:
I'm running the tests locally, it's taking a while. I'll check later.

Reply:
I've done the local tests

- Before:

```log
  Unsupported      :  686
  Passed           : 8957
  Expectedly Failed:   36
  Failed           :   38

4 warning(s) in tests
```

- After
```log
  Unsupported      :  686
  Passed           : 8957
  Expectedly Failed:   36
  Failed           :   38

4 warning(s) in tests
```